cpp/src/gandiva/projector.h

```diff
@@ -34,6 +34,34 @@ namespace gandiva {
 
 class LLVMGenerator;
 
+class ProjectorCacheKey {
```

Comment:
ProjectorCacheKey and FilterCacheKey are not really required now since we are now using a single cache and you are anyway creating basecachekey from them. Just create a single class named ExpressionCacheKey

Reply:
Done, now there is only the ExpressionCacheKey